src/components/cylc/tree/Tree.vue

```diff
@@ -17,21 +17,81 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 <template>
   <div class="my-2">
+    <v-layout
+      class="mb-1"
+    >
+      <v-flex
+        xs12
+        md5
+        class="mx-3">
+        <v-text-field
+            clearable
+            dense
+            flat
+            hide-details
+            outlined
+            placeholder="Filter by task name"
+            v-model="tasksFilter.name"
+        ></v-text-field>
+      </v-flex>
+      <v-flex
+          xs12
+          md5>
+        <v-combobox
+          :items="taskStates"
+          clearable
+          dense
+          flat
+          hide-details
+          multiple
+          outlined
+          placeholder="Filter by task state"
+          v-model="tasksFilter.states"
+        >
+          <template v-slot:item="slotProps">
+            <Task :status="slotProps.item.value.toLowerCase()" :progress=0 />
```

Comment:
Out of interest, I wouldn't have expected that we would need to convert task state to lower case?

Reply:
At the moment it's necessary. I can't recall exactly why, but the `TaskState` enum has values like `static SUBMITTED = new TaskState('SUBMITTED')`. Where the `String` value is set to `SUBMITTED`.

Here the component will use the `item.value` (`SUBMITTED`), which wouldn't match the CSS class.

![image](https://user-images.githubusercontent.com/304786/91500111-eba17200-e916-11ea-9285-a493772816b0.png)

I think we will have to review that enum, as there are task states in other parts of the code that are not using it yet, and I **think** there is some discussion somewhere about changing the states... or was it organizing jobs/tasks states?

Reply:
Ah I see, thanks for clearing that up.

I had a branch on Cylc Flow which converted task states to an enum which got exposed to GraphQL which would probably have made life easier, but unfortunately, serialising it for the DB/protobuf was a bit of a pain.